### PR TITLE
[SDK/CLI] Add pfazure run cancel

### PR DESCRIPTION
**Pull request by 0mza987**
# Description

Please add an informative description that covers that changes made by the pull request and link all relevant issues.

# All Promptflow Contribution checklist:
- [ ] **The pull request does not introduce [breaking changes].**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [ ] **I have read the [contribution guidelines](../CONTRIBUTING.md).**
- [ ] **Create an issue and link to the pull request to get dedicated review from promptflow team. Learn more: [suggested workflow](../CONTRIBUTING.md#suggested-workflow).**

## General Guidelines and Best Practices
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### Testing Guidelines
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: src/promptflow/tests/sdk_cli_azure_test/e2etests/test_run_operations.py [hunk 1]
```diff
@@ -417,6 +418,24 @@ def test_update_run(self, pf, randstr: Callable[[str], str]):
         assert run.description == new_description
         assert run.tags["test_tag"] == test_mark
 
+    def test_cancel_run(self, pf, runtime: str, randstr: Callable[[str], str]):
```
Comment (zhengfeiwang) [inline src/promptflow/tests/sdk_cli_azure_test/e2etests/test_run_operations.py:421]: no recording? is there any error you met with this operation? maybe I can help take a look?

Comment (0mza987) [reply src/promptflow/tests/sdk_cli_azure_test/e2etests/test_run_operations.py:421]: Simply forgot the add lol. Seems no error happened. Updated the recording.

**Review by yigao-alter — APPROVED** [review]

**Review by crazygao — DISMISSED** [review]

**Review by zhengfeiwang — APPROVED** [review]

File: src/promptflow/promptflow/_cli/_pf_azure/_run.py [hunk 2]
```diff
@@ -133,17 +131,20 @@ def add_parser_run_stream(subparsers):
...
+    ] + base_params
+
+    activate_action(
+        name="stream",
         description="A CLI tool to stream run logs to the console.",
         epilog=epilog,
-        help="Stream run logs to the console.",
+        add_params=add_params,
+        subparsers=subparsers,
+        help_message="Show a run.",
```
Comment (wangchao1230) [inline src/promptflow/promptflow/_cli/_pf_azure/_run.py:145]: why change the help message for stream?

Comment (0mza987) [reply src/promptflow/promptflow/_cli/_pf_azure/_run.py]: Good catch, was refactoring the stream parser functio. Recovered the help message.

**Review by crazygao — APPROVED** [review]

**Review by wangchao1230 — APPROVED** [review]

**Pull request merged**